eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -0,0 +1,133 @@
+            echo "##vso[task.setvariable variable=packageLocation]core/${{artifact.name}}"
+          }
+          if ('${{parameters.ServiceDirectory}}' -eq 'cognitiveservices') {
+            echo "##vso[task.setvariable variable=packageLocation]cognitiveservices/${{artifact.directoryName}}"
+            echo "##vso[task.setvariable variable=artifactSafeName]${{artifact.directoryName}}"
+          }
+          if ('${{parameters.LibType}}' -eq 'Management') {
+            echo "##vso[task.setvariable variable=packageLocation]${{artifact.serviceDir}}/${{artifact.name}}"
+            echo "##vso[task.setvariable variable=artifactSafeName]${{artifact.name}}"
+          }
+        displayName: Set Variables for current Loop
+      - pwsh: |
+          mkdir $(artifactSafeName)/dll-docs/my-api
+          mkdir $(artifactSafeName)/dll-docs/dependencies/my-api
+          mkdir $(artifactSafeName)/dll-xml-output
+          mkdir $(artifactSafeName)/dll-yaml-output
+          mkdir $(artifactSafeName)/docfx-output
+        displayName: Create Directories for Doc Generation
+        workingDirectory: $(Build.BinariesDirectory)
+      - pwsh: dotnet build eng/service.proj /p:ServiceDirectory=$(packageLocation) /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath=$(Build.BinariesDirectory)/$(artifactSafeName)/dll-docs/my-api /p:TargetFramework=netstandard2.0
+        displayName: Build Client Packages
+        condition: and(succeeded(), eq(False, '${{ parameters.LibType }}'))
+      - pwsh: dotnet build eng/service.proj /p:ServiceDirectory=$(packageLocation) /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath=$(Build.BinariesDirectory)/$(artifactSafeName)/dll-docs/dependencies/my-api /p:TargetFramework=netstandard2.0 /p:CopyLocalLockFileAssemblies=true
+        displayName: Build Client Packages, Include Dependencies
+        condition: and(succeeded(), eq(False, '${{ parameters.LibType }}'))
+      - script: dotnet msbuild eng/mgmt.proj /p:scope=$(packageLocation) /p:OutputPath=$(Build.BinariesDirectory)/$(artifactSafeName)/dll-docs/my-api -maxcpucount:1 -nodeReuse:false
+        displayName: Build Mgmt Packages
+        condition: and(succeeded(), eq(True, '${{ parameters.LibType }}'))
+      - script: dotnet msbuild eng/mgmt.proj /p:scope=$(packageLocation) /p:OutputPath=$(Build.BinariesDirectory)/$(artifactSafeName)/dll-docs/dependencies/my-api /p:CopyLocalLockFileAssemblies=true -maxcpucount:1 -nodeReuse:false
+        displayName: Build Mgmt Packages, Include Dependencies
+        condition: and(succeeded(), eq(True, '${{ parameters.LibType }}'))
+      - task: DeleteFiles@1
+        displayName: "Delete files from $(Build.BinariesDirectory)/$(artifactSafeName)/dll-docs/my-api"
+        inputs:
+          SourceFolder: "$(Build.BinariesDirectory)/$(artifactSafeName)/dll-docs/my-api"
+          Contents: |
+            *.json
+            *.pdb
+            *.config
+      - task: CmdLine@2
```

Comment:
Why are sometimes you using built-in task like DeleteFiles/CmdLine and sometimes you are calling script tasks and sometimes you are calling pwsh. It seems like you could use pwsh for all the calls. 

Also as a high-level I hope I never have to replicate these steps locally as there are so many it would be very difficult. We should consider making one ps script that wraps most of these steps up into one and that would make it easier to repo locally and probably easier to follow the logic. 

We should file a post-GA clean-up task for that work.

Reply:
Ok, I'll do that.

Reply:
Issue filed [here](https://github.com/Azure/azure-sdk-for-net/issues/8461)